src/test/ReactTestUtils.js

```diff
@@ -309,6 +309,44 @@ var ReactTestUtils = {
     return new ReactShallowRenderer();
   },
 
+  jasmineMatchers: {
+    /**
+    * Expect function to perform a console.warn.
+    * If `expectation` is a number, we expect the function to make `expectation` calls to warn.
+    * If `expectation` is a string, we expect that string to appear in the warn output.
+    **/
+    toWarn: function(expectation) {
+      mocks = require('mocks');
+      var warn = console.warn;
+      console.warn = mocks.getMockFunction();
+      try {
+        this.actual();
+        var consoleLog = console.warn.mock.calls;
+        var result = { pass: false };
+        if(typeof(expectation) == 'number') {
```

Comment:
style nit: space after keywords (`if`) also,  no parens with `typeof`. also also, `===` not `==`